arches/settings_utils.py

```diff
@@ -121,38 +129,63 @@ def build_templates_config(
     ]
 
 
-def transmit_webpack_django_config(
-    root_dir,
-    app_root,
-    static_url,
-    public_server_address,
-    webpack_development_server_port,
-    arches_applications=None,
-):
-    arches_applications_paths = {}
+def inject_arches_applications_directories():
```

Comment:
The logic in this function doesn't hold up to all the conditional cases in the other `settings_utils` functions whose logic it's replacing. Can they not exist unaltered?

Reply:
Happy to walk through my thought process, this part is still pretty request-for-comment.

There are three places in settings.py where we need information about arches applications. Two of them are in the settings themselves, in other words, in the middle of creating a constant in settings.py we call a settings util to inject stuff. The third place is at the bottom of settings.py where we transmit certain settings to nodejs to build the frontend.

The basic idea in this PR is to shift the burden of declaring arches applications to the arches application, using the standard Django AppConfig and app registry. That's not available until you run django.setup(), which requires you to know what your settings are, which was basically the first thing you and I both found when hacking on this -- we're caught in a bind where we want to dynamically alter settings based on the value of other settings/apps.

So the ideas I've been toying with here are non-invasive ways to call django.setup(). django.setup() assumes you have settings you can feed to it. So that's a perfect fit for that third settings.py location where we're feeding settings. It's not a great fit for the first two cases where we're basically injecting stuff into the assignment of a settings constant. A few options for those cases:

- forget the app registry/documented django interface, just directly import the module with importlib (pros: works, cons: hacky, doesn't account for the way django lets you define multiple configs, and it's figured out for you)
- manually create AppConfigs, kind of like the registry does (pros: kinda works, cons: might bork later callers of django.setup()?)
- defer the injecting of arches application templates/staticfiles information to the apps.py ready() method (pros: more idiomatic for django, cons: ...?)

This latest iteration of this PR is taking this last approach.

> The logic in this function doesn't hold up to all the conditional cases in the other settings_utils functions whose logic it's replacing.

This latest approach wasn't supposed to miss any logic, so if you can help point me to what I'm missing, I'd very appreciative 🙏

Reply:
ah fair, on second readthrough this makes sense